### PR TITLE
fix: clockout most recent half shift

### PR DESCRIPTION
**Pull request by Timbobeek**
### Bug
 
Clockout uses the last shift based on insert order in the database not based on time of the shift. 

### Fix

With this code, employee clockout will bring up the most recent (timewise) half-shift that the employee has.

### CODE REVIEW

File: desktop/src/store/Shift/selectors.jsx [hunk 1]
```diff
@@ -22,9 +22,15 @@ export const getCurrentShift = createSelector(
...
   getShiftFromState,
-  (shifts, results, shift) => {
+  getEmployeeFromState,
+  (shifts, results, employee) => {
+
     if (!results || results.length === 0) return null
-    return shifts[shift.current.id]
+
+    let employeesHalfShifts = Object.values(shifts).filter(s => s.employeeId === employee.current.id && s.clockOutDate === null);
+    let mostRecentHalfShift = employeesHalfShifts.sort((a,b) => { return ( new Date (b.clockInDate) - new Date (a.clockInDate) ) })[0];
```
Comment (joshuawootonn) [inline desktop/src/store/Shift/selectors.jsx:31]: Nice work!

**Review by joshuawootonn — APPROVED** [review]

File: desktop/src/store/Shift/selectors.jsx (hunk 1 above)
Comment (Timbobeek) [reply desktop/src/store/Shift/selectors.jsx:30]: It happened to be a partial fix, found rest of the problem, fixed in this PR.

**Review by Timbobeek — COMMENTED** [review]
Additional code fixes the issue completely.

**Review by Timbobeek — COMMENTED** [review]
Done

**Review by joshuawootonn — APPROVED** [review]

**Pull request merged**